crates/jstzd/tests/octez_node_test.rs

```diff
@@ -0,0 +1,77 @@
+use jstzd::task::{octez_node, Task};
+
+async fn retry<'a>(
+    retries: u16,
+    interval_ms: u64,
+    f: &(dyn Fn() -> std::pin::Pin<
+        Box<dyn std::future::Future<Output = anyhow::Result<bool>> + Send + 'a>,
+    > + Sync),
+) -> bool {
+    let duration = tokio::time::Duration::from_millis(interval_ms);
+    for _ in 0..retries {
+        tokio::time::sleep(duration).await;
+        if let Ok(v) = f().await {
+            if v {
+                return true;
+            }
+        }
+    }
+    false
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn octez_node_test() {
+    let data_dir = tempfile::tempdir().unwrap();
+    let log_file = tempfile::NamedTempFile::new().unwrap();
+    let port = std::net::TcpListener::bind("127.0.0.1:0")
+        .unwrap()
+        .local_addr()
+        .unwrap()
+        .port();
+    let rpc_endpoint = format!("localhost:{}", port);
+
+    let mut config_builer = octez_node::OctezNodeConfigBuilder::new();
+    config_builer
+        .set_binary_path("octez-node")
+        .set_data_dir(data_dir.path().to_str().unwrap())
+        .set_network("sandbox")
+        .set_rpc_endpoint(&rpc_endpoint)
+        .set_log_file(log_file.path().to_str().unwrap())
+        .set_run_options(&[]);
+    let mut f = octez_node::OctezNode::spawn(config_builer.build().unwrap())
+        .await
+        .unwrap();
+
+    let health_check_endpoint = format!("http://{}/health/ready", rpc_endpoint);
+    // Should be able to hit the endpoint since the node should have been launched
+    let node_ready = retry(10, 1000, &|| {
+        Box::pin(async {
+            let res = reqwest::get(&health_check_endpoint).await;
+            if let Ok(raw_body) = res {
+                let body = raw_body
+                    .json::<std::collections::HashMap<String, bool>>()
+                    .await
+                    .unwrap();
+                return body.get("ready").cloned().ok_or(anyhow::anyhow!(""));
+            }
+            Err(anyhow::anyhow!(""))
+        })
+    })
+    .await;
+    assert!(node_ready);
+
+    let _ = f.kill().await;
+    // Wait for the process to shutdown entirely
+    let node_destroyed = retry(10, 1000, &|| {
+        Box::pin(async {
+            let res = reqwest::get(&health_check_endpoint).await;
+            // Should get an error since the node should have been terminated
+            if let Err(e) = res {
+                return Ok(e.to_string().contains("Connection refused"));
+            }
+            Err(anyhow::anyhow!(""))
+        })
+    })
+    .await;
```

Comment:
With the change to ref taking an `impl Fn` closure, there shouldn't be a need to pass the closure as `&` or pin the async func. Should be able to do the following I think: 
```
retry(10, 1000, || async {
 ... 
});
```